### ISSUE TITLE
The extracted AEA key file is missing one

### ISSUE THREAD

**Issue by ssdswi** [blacktop/ipsw#570]
### What happened?

There are two *dmg.aea in IPSW file,but only can extract one key by *dmg.aea.pem.
The other unextracted key is the key for the most critical dmg file, which is the largest in file size.
![屏幕截图(169)](https://github.com/user-attachments/assets/4d5efc5f-3ed9-4cc4-be52-bdeed4a9a455)
![屏幕截图(170)](https://github.com/user-attachments/assets/b4885b9d-d63b-4652-bf08-ba78821ce77c)

### How can we reproduce this?

use ‘ipsw extract --fcs-key’

### ipsw version

```bash
Version: 3.1.544,
```

### Search

- [X] I did search for other open and closed issues before opening this

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ssdswi** [comment]
Oh,i'm sry.I found the fs and the sys fcs key are the same.It's my fault.
But i can't open any ios18 decrypted *.dmg file with “Transmac” on Windows. I can open the biggest ios17 and before *.dmg file in IPSWs at “transmac”APP in Windows.
That's why I made this mistake.
Is there any other way to open it on Windows?

**Comment by blacktop** [comment]
do you have apfs-fuse installed?

**Comment by blacktop** [comment]
if so you could try: `ipsw mount fs IPSW.ipsw`

**Comment by ssdswi** [comment]
I'm using the apfs-fuse project, which seems to only work in a Linux environment, so I tried to mount the dmg using "ipsw mount" in a Linux environment, but it seems that ipsw doesn't call my compiled apfs-fuse directly even though I've added it to the $PATH.
![屏幕截图(173)](https://github.com/user-attachments/assets/183b754f-1a44-4bed-869c-440b4b0bc7b5)

So I tried extracting the dmg file and then mounting it directly using "apfs-fuse" and it seems to work. So in fact the decrypted dmg file should be correct, but there doesn't seem to be a way to use the file inside the dmg directly on Windows at the moment, but it should work fine on Linux or macos!
![屏幕截图(174)](https://github.com/user-attachments/assets/4bbbd8d2-15fd-4fff-ba58-068bef67d3a6)

So how exactly should I use "ipsw mount" directly within the ipsw project without prompting for the apfs command not found? Or is it possible to fully port this project to Windows?

**Comment by blacktop** [comment]
that is strange, I wonder if you can all apfs-fuse from anywhere (not just in the 'buid' folder to show it really can be found in your PATH?

I'm going to add a PATH lookup to then use the absolute path, let me know if that fixes anything for you too?

**Comment by aman77724** [comment]
Doss this issue resolved or is there another way to extract that big .dmg.aea file .

**Comment by Hamm0udx** [comment]
> do you have apfs-fuse installed?

Please make a video with the steps. I am a big fan of your work, but I did not understand the method.

**Comment by blacktop** [comment]
What version of Go are you using? And are you running your test inside of docker?

**Comment by MicahMartin** [comment]
> What version of Go are you using? And are you running your test inside of docker?

`go version go1.23.9 linux/amd64`
im not running inside dockerk

**Comment by MicahMartin** [comment]
> `ipsw` is built with latest 1.24 Go, also I assume this is running on Linux (the failure) and not in Docker?
> 
> That makes no sense to me as it's doing that same thing:
> 
> [ipsw/internal/utils/macos.go](https://github.com/blacktop/ipsw/blob/25c2444b16bbc9ea62d43f6750dcdbc60bbce86d/internal/utils/macos.go#L478)
> 
> Line 478 in [25c2444](/blacktop/ipsw/commit/25c2444b16bbc9ea62d43f6750dcdbc60bbce86d)
> 
>  apfsFusePath, err := execabs.LookPath("apfs-fuse")

yeah its really weird.. It must be something on my system? (ubuntu 24.04) I'm pulling down the repo now to see if I can replicate it with a build. I'm currently using the snap version

**Comment by MicahMartin** [comment]
Im looking into it now, turning my local ipsw into a snap repo & running it locally with strict confinement. and if it is the case, I'll look into using https://snapcraft.io/docs/interfaces - this is just my opinion but if you could avoid having users specify --classic when installing that'd cleaner fair to say?

**Comment by blacktop** [comment]
That sounds good to me.  I'm 💯 🍎 so I don't use any of the linux installers, but like to support ALL the pkg mgrs and make it as easy as possible for people to install run `ipsw` on their platform of choice.  I want to protect their security and privacy in any ways that I can, so I set up the snap pipeline w/ hopefully good defaults.  If there are better settings I am open to that.

**Comment by MicahMartin** [comment]
https://github.com/blacktop/ipsw/pull/723
